### ISSUE TITLE
Schema for BigDecimal is not correct

### ISSUE THREAD

**Issue by Flowneee** [GREsau/schemars#123]
## Problem

`0.8.8` version added schema for `BigDecimal`, but it is incorrect, because BigDecimal serialized as string, and can be deserialized from string, number or integer.

## Details

Current schema
```json
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "title": "Number",
  "type": "number"
}
```

For example:
```rust
use bigdecimal::BigDecimal;
use schemars::schema_for;

fn main() {
    let schema = schema_for!(BigDecimal);
    println!("{}", serde_json::to_string_pretty(&schema).unwrap());
    println!(
        "{:?}",
        serde_json::to_value(&BigDecimal::default()).unwrap()
    )
}
```

gives

```
Schema: {
  "$schema": "http://json-schema.org/draft-07/schema#",
  "title": "Number",
  "type": "number"
}
Serialized: String("0")
```

see https://www.jsonschemavalidator.net/s/s2ULf39e as well.

And about deserialization:

```rust
use bigdecimal::BigDecimal;

fn main() {
    assert!(serde_json::from_str::<BigDecimal>("123").is_ok());
    assert!(serde_json::from_str::<BigDecimal>("123.123").is_ok());
    assert!(serde_json::from_str::<BigDecimal>("\"123\"").is_ok());
}
```

## Solution

I don't have any _good_ solutions. The problem is: this type have different schemas for deserialization and serialization. One possible solution: use most common type here - string and then add description and pattern to it.

I recently faced same problem in my work project, and we came up with 2 wrapper types, which have different `JsonSchema` implementations, but this is quite ugly and would not be suitable for this lib I think.

**Comment by aldofunes** [comment]
I was able to make it work by mapping a BigDecimal to an `anyOf` config, where one is a number and the other a string with a regex that validates it's actually a number.

I ran some tests and it worked great for my use case, and thought that it was best to share it, and thus I made PR 224. Hope it helps

**Comment by GREsau** [comment]
This is mostly fixed in v0.8.14, in that the schema will now match how `BigDecimal` is serialised (i.e. as a numeric string). There's not yet a good way to specify different schemas for serialization vs deserialization (#48) though, so the schema doesn't accept a plain number even though it would be valid for deserialization

**Comment by SohumB** [comment]
Just as a note — we use rust_decimal schemas in our input jsonschemas, and we currently expect schemars `min` and `max` annotations to work on them. with this change, they no longer work, so we're currently pinning to v0.8.13 pending further investigation. I'm not intending this as a bug report exactly, just more of a usage note.
